samples/transactional-session/cosmosdb/sample.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Using Transactional Session with CosmosDB
+reviewed: 2022-07-20
+component: Core
+related:
+- nservicebus/messaging/transactionalsession
+---
+
+This sample uses the transactional session package with the CosmosDB persistence package to achieve transactionally consistent database changes and message operations.
+
+downloadbutton
+
+## Overview
+
+The sample contains a frontend and a backend service sharing a CosmosDB database instance. When the transactional session on the frontend is commited, an `Order` document is created with the status `Received` and an `OrderReceived` event is published. The backend service subscribes to the event and loads the order document to update it's status to `Accepted`.
+
+```mermaid
+sequenceDiagram
+    Frontend->>ITransactionalSession: Commit()
+    activate ITransactionalSession
+    ITransactionalSession->>CosmosDB: Store(order)
+    ITransactionalSession->>Queue: Publish(orderReceived)
+    ITransactionalSession-->>Frontend: 
+    deactivate ITransactionalSession
+    Queue->>Backend: Process(orderReceived)
+    activate Backend
+    Backend->>CosmosDB: Update(order)
+    Backend-->>Queue: 
+    deactivate Backend
+```
+
+The database and transport operations are executed atomically when the session is committed. If the session is aborted all database and transport operations are rolled back.
+
+## Prerequisties
+
+The sample is intended to be used with the [CosmosDB emulator](https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator?tabs=ssl-netstd21) to run locally. Alternatively, a connection string to an Azure CosmosDB instance can be provided.
+
+
+## Configuration
+
+### Frontend
+
+The `Frontend` service needs to enable the transactional storage feature in the endpoint configuration:
+
+snippet: cosmos-txsession-frontend-config
+
+Note that the `Outbox` feature must be enabled for the transactional session feature to work.
```

Comment:
Should we elaborate more on this? Or do we leave the batched scenario exclusively for the docs?

Reply:
I think that's for other docs